artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -171,15 +165,15 @@ public static AMQPMessage fromCore(ICoreMessage coreMessage, StorageManager stor
             properties.setMessageId("ID:" + message.getInnerMessage().getUserID().toString());
          }
       }
-      Destination destination = message.getJMSDestination();
+      SimpleString destination = message.getJMSDestination();
       if (destination != null) {
-         properties.setTo(toAddress(destination));
-         maMap.put(JMS_DEST_TYPE_MSG_ANNOTATION, AMQPMessageSupport.destinationType(destination));
+         properties.setTo(destination.toString());
```

Comment:
Wheres the original logic that was being done inside toAddress

Reply:
Same that I said before...

Reply:
I dont follow

Reply:
there's a bug on this part, I'm fixing it.

Reply:
Can we revert master? And then review this properly?

Reply:
@michaelandrepearce I did not want to depend on JMS at the first time I implemented the converter. I did it back then because it was the easy route. (copy and paste the client API, depend on the interfaces and get done with it).

Now I need to remove the dependency. Using concrete classes is the way to go here.. I don't think there's much to discuss. .unless everybody is opposing to this change.. which I can't see it as a bad thing.